### PR TITLE
Fix rpc method inconsistencies for deposit txs.

### PR DESCRIPTION
**Pull request by kevinhalliday**
Running branch of rpc fixes for issues found in rpc tests. No need to merge yet.

### CODE REVIEW

File: rpc/backend/tx_info.go [hunk 1]
```diff
@@ -310,7 +324,7 @@ func (b *Backend) GetTransactionReceipt(hash common.Hash) (map[string]interface{
 
 			// Inclusion information: These fields provide information about the inclusion of the
 			// transaction corresponding to this reciept
-			"blockHash":        txDetails.TxReceipt.TxHash,
+			"blockHash":        common.BytesToHash(block.Block.Header.Hash()).Hex(),
```
Comment (percival7777) [inline rpc/backend/tx_info.go:327]: No need to fetch the block again just use `txDetails.TxReceipt.BlockHash`, I think this was a typo on my end.

**Review by percival7777 — CHANGES_REQUESTED** [review]

File: rpc/backend/tx_info.go (hunk 1 above)
Comment (percival7777) [inline rpc/backend/tx_info.go]: No need to fetch the block again just use `txDetails.TxReceipt.BlockHash`, I think this was a typo on my end.

**Review by percival7777 — CHANGES_REQUESTED** [review]

File: rpc/backend/tx_info.go [hunk 2]
```diff
@@ -66,9 +68,12 @@ func (b *Backend) getRPCTransactionForEndBlockTx(txDetails *EndBlockTxDetails) (
 		Type:             2,
 		Accesses:         nil,
 		ChainID:          (*hexutil.Big)(b.chainID),
-		V:                nil,
-		R:                nil,
-		S:                nil,
+		// V:                nil,
```
Comment (skosito) [inline rpc/backend/tx_info.go]: can we remove these comments?
what is difference between having nil and 0 here?

**Review by skosito — APPROVED** [review]

File: rpc/backend/tx_info.go (hunk 2 above)
Comment (kevinhalliday) [reply rpc/backend/tx_info.go]: yes. this was ST geth could unmarshal tx objects on client. before it would error out. I can add a comment explaining

**Pull request merged**